packages/remix-react/routeModules.ts

```diff
@@ -190,10 +190,21 @@ export async function loadRouteModule(
     routeModulesCache[route.id] = routeModule;
     return routeModule;
   } catch (error: unknown) {
-    // User got caught in the middle of a deploy and the CDN no longer has the
-    // asset we're trying to import! Reload from the server and the user
-    // (should) get the new manifest--unless the developer purged the static
-    // assets, the manifest path, but not the documents 😬
+    // If we can't load the route it's likely one of 2 things:
+    // - User got caught in the middle of a deploy and the CDN no longer has the
+    //   asset we're trying to import! Reload from the server and the user
+    //   (should) get the new manifest--unless the developer purged the static
+    //   assets, the manifest path, but not the documents 😬
+    // - Or, the asset trying to be imported has an error (usually in vite dev
+    //   mode), so the best we can do here is log the error for visibility
+    //   (via `Preserve log`) and reload
+
+    // Log the error so it can be accessed via the `Preserve Log` setting
+    console.error(
+      `Error loading route module \`${route.module}\`, reloading page...`
+    );
+    console.error(error);
```

Comment:
We've never logged this error before since it's swallowed on the reload and (I think) it was much harder to run into this for import issues in dev when using `esbuild`.  But using vite during dev, you don't find out about these module errors until runtime, so we can log them and they're available via `Preserve log`

I'm curious through if they could be surfaced through Vite's overlay? @pcattori @markdalgleish